modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -46,21 +48,23 @@ const nestedBlockMerge = (
   }
 
   const position = CaretFinder.prevPosition(toBlock.dom, CaretPosition.before(insertionPoint.dom));
-  Arr.each(extractChildren(fromBlock), (child) => {
+  Arr.each(extractChildren(fromBlock, schema), (child) => {
     Insert.before(insertionPoint, child);
   });
   removeEmptyRoot(rootNode, fromBlock);
   return position;
 };
 
-const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
+const isInline = (schema: Schema) => (node: SugarElement<Node>): node is SugarElement<HTMLElement> => schema.isInline(SugarNode.name(node));
```

Comment:
Does this need to be curried? The only call I see the line where it's being called like isInline(schema)(child)?

Reply:
it isn't needed I changed it